lib/SPIRV/OCLUtil.cpp

```diff
@@ -145,6 +145,16 @@ template <> void SPIRVMap<OCLScopeKind, Scope>::init() {
   add(OCLMS_sub_group, ScopeSubgroup);
 }
 
+template <> void SPIRVMap<std::string, Scope>::init() {
+  add("work_item", ScopeInvocation);
+  add("workgroup", ScopeWorkgroup);
```

Comment:
Should this be "workgroup" or "work_group"?  In atomic_syncscope_test.ll the comment on line 23 has "work_group".

Reply:
From:  https://www.khronos.org/assets/uploads/developers/presentations/opencl20-quick-reference-card.pdf

> The default scope is work_group for local atomics and all_svm_devices for global atomics.

Reply:
I actually don't mind adding both in sake of usability

Reply:
> "workgroup" or "work_group"

IMHO we should target what clang can generate, and at this point of time it's 'workgroup'. And thanks for noticing default scope in the spec!

Reply:
We should only accept what the spec says.  If "workgroup" is meaningless according to the spec, it should be treated the same as "wrong_scope".  Personally, I would prefer if unrecognized scope strings were treated as errors.

Reply:
So, am I right that all the scopes except "device" should go with underscore? I'm not sure if clang generates such things for now though

Reply:
>> Personally, I would prefer if unrecognized scope strings were treated as errors.

While it is indeed better - I would still recommend not to threat them as errors as we and community will face unknown number of regressions because of it, so I'd preserve the current behavior with the defaulting it if the scope is unknown.

Reply:
I suggest to align namings (if needed) and fix default scopes in a separate PR - need to investigate why different scopes were chosen by default earlier, find which scopes clang is generating now, and analyze how the changes would affect e2e tests.